### PR TITLE
Add Docker caching

### LINKED ISSUE #31: Add Docker caching

### PR DESCRIPTION
**Pull request by maxslarsson**
Fixes #31

### CODE REVIEW

**Comment by maxslarsson** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/29?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#29** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/29?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maxslarsson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .gitignore
Comment (ZachGarcia42) [inline .gitignore:1]: Was this an accident?

**Review by ZachGarcia42 — COMMENTED** [review]
Fix merge conflicts please

**Review by ZachGarcia42 — CHANGES_REQUESTED** [review]
Fix merge conflicts please

**Review by ZachGarcia42 — COMMENTED** [review]
Are we able to just rerun the workflow to see a speed increase from caching?

@ZachGarcia42 reviewed this PR from Slack with [Graphite](https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/29)

**Comment by maxslarsson** [comment]
Yea we can, but I don't think it'll have a huge speed increase unfortunately because of how Docker works. Basically it treats each line in the Dockerfile as a layer, so whenever the inputs for one line changes, it has to rebuild every line below. So whenever we change any code, it invalidates the `COPY . .` line, which means we have to rerun each line below which includes the `bazel build` line. That means I don't think it will get cached. Ideally, we'd be able to build the dependencies for our program on a seperate line before the `COPY . .`, so that that gets cached and the `COPY . .` is put after.

Do you know if there is a Bazel command to download and build the dependencies only?

**Comment by ZachGarcia42** [comment]
I don't think that's too feasible unfortunately. 13 minutes isn't terrible. If that's slower than we want, we can create a second docker file for the workflow and move back to uploading/downloading an artifact from the bazel tests

**Comment by maxslarsson** [comment]
Yea for sure, I think 13 minutes is fast enough for our purposes, at least for now. If not, we can go back in history and get your artifact solution again.

I think most of the time we test the code locally, and the Docker build is just to get onto the RPi

**Review by ZachGarcia42 — APPROVED** [review]
Thanks for this improvement!

**Pull request merged**